Disable endpoint connectivity health checking in chaining mode

tgraf opened the pull request:
Chaining mode means that we can't allocate IPs and thus the endpoint connectivity health checking must be disabled.

Fixes: #9260 
Fixes: #8743

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cilium/cilium/9460)
<!-- Reviewable:end -->

tgraf committed:
helm: Disable endpoint-health-checking when chaining is enabled

Signed-off-by: Thomas Graf <thomas@cilium.io>

tgraf committed:
agent: Add --enable-endpoint-health-checking flag

Add ability to have node level connectivity health checking while disabling
virtual endpoint connectivity health checks.

Signed-off-by: Thomas Graf <thomas@cilium.io>

aanm commented on Documentation/cmdref/cilium-agent.md:
```diff
@@ -55,6 +55,7 @@ cilium-agent [flags]
       --egress-masquerade-interfaces string                   Limit egress masquerading to interface selector
+      --enable-endpoint-health-checking                       Enable connectivity health checking between virtual endpoints (default true)
       --enable-endpoint-routes                                Use per endpoint routes instead of routing via cilium_host
       --enable-health-checking                                Enable connectivity health checking (default true)
```
Can you change this one to ` Enable connectivity health checking between nodes (default true)`

tgraf replied:
@aanm The flag will disable health checking entirely right now.

tgraf committed:
flannel: Disable endpoint connectivity health check

No reason to disable endpoint health checking overall. Disabling endpoint
connectivity health checking is sufficient.

Signed-off-by: Thomas Graf <thomas@cilium.io>

aanm commented on daemon/ipam.go:
```diff
@@ -194,7 +194,7 @@ func (d *Daemon) allocateDatapathIPs(family datapath.NodeAddressingFamily) (rout
 func (d *Daemon) allocateHealthIPs() error {
 	bootstrapStats.healthCheck.Start()
-	if option.Config.EnableHealthChecking {
+	if option.Config.EnableEndpointHealthChecking {
```
this should then be `option.Config.EnableHealthChecking && option.Config.EnabledEndpointHealthChecking` because one disables the other.

tgraf replied:
Why? The health IP is only needed if we perform endpoint health checking. When health checking is enabled and endpoint health checking is disabled, no health IP is required. All health checking will be done across node IPs.

aanm replied:
>The health IP is only needed if we perform endpoint health checking.

Correct

>When health checking is enabled and endpoint health checking is disabled, no health IP is required.

Also correct.

Why should any kind of health checking, including endpoint health checking, be enabled if `EnableHealthChecking` is set to false? That's why I said that the if statement should be `option.Config.EnableHealthChecking && option.Config.EnabledEndpointHealthChecking`

tgraf replied:
Sorry, now I understand what you mean. You are absolutely right.

ianvernon commented on Documentation/cmdref/cilium-agent.md:
```diff
@@ -55,6 +55,7 @@ cilium-agent [flags]
       --egress-masquerade-interfaces string                   Limit egress masquerading to interface selector
+      --enable-endpoint-health-checking                       Enable connectivity health checking between virtual endpoints (default true)
       --enable-endpoint-routes                                Use per endpoint routes instead of routing via cilium_host
       --enable-health-checking                                Enable connectivity health checking (default true)
```
Does this option define purely node-to-node connectivity health checking now? What differentiates this between `--enable-endpoint-health-checking`? I feel like to end-users this would not be very clear.

tgraf replied:
The existing `--enable-health-checking` remains unchanged in functionality and disables health checking overall. I did not want to change the behavior of an existing flag.
